### ISSUE TITLE
Redefine CAT and Nekoin token.

### ISSUE THREAD

**Issue by hurui200320** [NekoHitDev/whitepaper#2]
Today I'm working on the `v1-RC1` version of the whitepaper and noticed something about the token.

## CatToken

The CAT Token should be used to support the content creator, so I want to eliminate the influence of marketing fluctuation. Otherwise, they all will buy the coin at a low price and sell them when high, and no one would supporting creators anymore. We can create a reasonably stable currency by fixing the exchange rate and opening the ability to issue new coins to everyone. Here is the detailed plan:

+ We chose a stable coin, which might be a wrapper of USDT/BUSD/something on N3, and we fixed the exchange rate: 1 CAT = 0.5 USD.
+ User can transfer that wrapper coin to CatToken contract. The contract will issue tokens and add them to the user's address. 
+ User can destroy some amount of CAT token, then CatToken contract will transfer the wrapper coin to the user's address.

This will make the CatToken become a wrapper of a wrapper coin, and it might be abandoned when we implement multiple coins accepted by WCAContract. But considering we don't have that right now, it's a good placeholder. Also, in the future, we might be able to introduce some new features to CatToken.

## Nekoin

The Nekoin is used for the DAO of this project. And it should be used for voting. Thus we can have a variable total supply amount and issue and destroy tokens in the future. This coin doesn't care about its value. Just like Neo, holding it means you are a community member of this project. You can trade them for other coins, but it's not part of our design.

----

The define of those tokens is open to discussion. Feel free to reply to this issue.

**Comment by hurui200320** [comment]
If we cannot find a wrapper coin on the N3 mainnet by the end of October (according to the plan, we are going to test the released code on testnet with our community members by the end of Oct.), then we can fall back to the current implementation of CatToken but reduce the total supply. Otherwise, there will be a big issue when we migrate to the fixed exchange rate after doing the air drop.

**Comment by ykoitzsch** [comment]
I only know of flamingo that they offer wrapped coins from other chains but if I remember their roadmap, it wont be live at the end of october. Also if we use wrapped coins we do have a dependency on them which can be bad (but doesnt have to be). Also the lyrebird project will go live only next year in Q1 so for now I feel like we have to stick with the original plan.

**Comment by ykoitzsch** [comment]
instead of using a wrapper for a stable coin we could make use of an oracle and let users exchange their GAS/NEO for CAT tokens at a specific exchange rate to get the same results

**Comment by ykoitzsch** [comment]
but oracle brings some other problems: high GAS fee when exchaning, more vulnerable to exploits..

**Comment by hurui200320** [comment]
> Also if we use wrapped coins we do have a dependency on them which can be bad

If we are stick with one special wrapped coin, we have to fully trust it, which I feel not good. The USDT's price dropped a lot (for a stable coin) when the news said they hold EverGrande's securities/stock/whatever it is. But using multiple stable coins will introduce additional work when users exchange coins.

> but oracle brings some other problems: high GAS fee when exchaning, more vulnerable to exploits..

Oracle is kind of fragile: We need some sources to get the exchange rate of a given coin pair, which means we have to rely on third-party services and interact with real-world things. I guess it will increase the vulnerability if we fail to take excellent care, which reminded me of the fail of "the DAO" on ETC. I don't want to be "the DAO" on Neo N3.

----

But in general, I think "opening the ability to issue new coins to everyone" is not a bad idea. So I want to continue this idea, but with limited resources/products on N3, we can do that later, as the original plan. But I want to address a point:

> We have to pay for the airdrop when we make CatToken's value stable. If we cannot implement the exchange with stable coins now, we might need to reduce the total supply of CatToken. But the question is, how much is the new total supply amount?

Initially, I thought more supply will do the trick, but actually no. After several months of watching coin prices on Binance, I noticed every coin price would fluctuate as long as it's not a stable coin. I'm kinda naive at that time.

I talked with @Wakayume , he will conduct the airdrop event. The rough idea is to invite some content creators and let them create some projects on our platform, and we will cover the staked fee. Then we airdrop some CatToken to their reader/supporters. During that process, we will clearly declare that CatToken has a fixed exchange rate and exchange will be implemented in the future.

Also, I think use wrapped coins is easier to do. But we can "let the community decides"™ when there are several stable/wrapped coins on N3.
